File: java-takeaway-challenge/app/src/main/java/OrderSystem/Menu.java
package OrderSystem;

import java.util.ArrayList;

public class Menu {
  private ArrayList<Dish> menu = new ArrayList<Dish>();

  public ArrayList<Dish> getMenu() {
    return menu;
  }

  public void addDish(Dish dish) {
    menu.add(dish);
  }

  public void removeDish(Dish dish) {
    menu.remove(dish);
  }
}

File: golden-square-solo-project/lib/menu.rb
require 'dish'

class Menu
  def initialize
    @dishes = []
  end

  def add_dish(dish)
    fail 'Invalid dish data' unless dish.name.is_a?(String) && dish.price.is_a?(Float)
    @dishes << dish
  end

  def remove_dish(dish)
    @dishes.delete(dish)
  end

  def get_menu
    @dishes
  end
end